torch/csrc/Module.cpp

```diff
@@ -1258,6 +1259,29 @@ Call this whenever a new thread is created in order to propagate values from
   py_module.def("_dispatch_key_set", [](const at::Tensor& x) {
     return toString(x.key_set());
   });
+  py_module.def(
+      "_change_backend_component_keys", [](at::Tensor& x, c10::Device device) {
+        x._change_backend_component_keys(device);
+      });
+  py_module.def("_add_meta_to_tls_dispatch_include", []() {
+    auto local_keyset = c10::impl::tls_local_dispatch_key_set();
+    c10::DispatchKeySet key_set({at::DispatchKey::Meta});
+    local_keyset.included_ = local_keyset.included_ | key_set;
+    c10::impl::_force_tls_local_dispatch_key_set(local_keyset);
+  });
+  py_module.def("_remove_meta_from_tls_dispatch_include", []() {
+    auto local_keyset = c10::impl::tls_local_dispatch_key_set();
+    c10::DispatchKeySet key_set({at::DispatchKey::Meta});
+    auto k = key_set.highestBackendKey();
+    local_keyset.included_ = local_keyset.included_.remove_backend(k);
+    c10::impl::_force_tls_local_dispatch_key_set(local_keyset);
+  });
```

Comment:
feels like it would be better to bind one of the local TLS RAII objects. It's pretty quick and easy

Reply:
I'm trying to fix the functionalization <> dynamic shape issues and I ended up adding something pretty similar here: [_change_backend_component_keys.](https://github.com/pytorch/pytorch/pull/82453/files#diff-b3651b15177d065d4a02b0bd03703b6df569fc7a53ce88c4c6dbd6145adf35f6R89). If this looks good to you, then we can either stamp those changes into this PR or fix this code up after my PR lands.

Reply:
If you don't mind fixing up this code after my PR lands, that would be great. Thanks brian.